connectors/sources/mongo.py

```diff
@@ -191,10 +190,24 @@ def get_default_configuration(cls):
             },
         }
 
+    async def _get_client(self):
+        try:
+            self.client = AsyncIOMotorClient(self.host, **self.client_params)
+            self.db = self.client[self.configuration["database"]]
+            self.collection = self.db[self.configuration["collection"]]
+        except InvalidURI as exception:
+            msg = f"Found Invalid URL, please check configuration. Error: {exception}"
+            raise InvalidURI(msg) from exception
+        except Exception as exception:
+            msg = f"Somthing went wrong while creating a client. Error: {exception}"
+            raise Exception(msg) from exception
```

Comment:
I advice, if possible, to make client a context - so it could be used with something like:

```
with self.get_client() as client:
    # call MongoDB
```

Otherwise the problems are:

1. What if you `_get_client` twice? First one will be overwritten and never disposed of
2. Hard to track who created client, cause it could have been created in different places.
3. You have to check in all places `if self.client is None`